### PR TITLE
Use GitHub actions for continuous integration (CI)

### PR DESCRIPTION
**Pull request by SimonFrings**
This pull requests updates this project to use GitHub actions for continuous integration instead of travis. I tried to avoid tackling too many topics at once in here, so I'll look into supporting newer PHP versions and updating the documentation in a follow up PR. This is an internal refactoring only and shouldn't have any impact on projects using this as a dependency.

I also saw that #10 and #14 were working on similar topics, so my pull request here will supersedes them both and closes them once this PR gets merged.

Closes #14

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,17 @@
...
         "php": ">=5.3"
     },
+    "require-dev": {
+        "phpunit/phpunit": "^4.8.36"
+    },
     "autoload": {
-        "psr-0": { "Clue\\Redis\\Protocol": "src" }
+        "psr-4": {
+            "Clue\\Redis\\Protocol\\": "src/Clue/Redis/Protocol/"
+        }
```
[the remark this answers is not in the document]
Comment (clue) [reply composer.json:22]: Changing to PSR-4 make sense to me, but looks like this change is best discussed in a separate PR (#10), no? (See also `autoload-dev` and namespace additions)

[the remark this answers is not in the document]
Comment (SimonFrings) [reply composer.json]: Sounds good to me, I updated the PR accordingly. Not sure if #10 will be continued, so I might create a follow-up PR that overtakes #10 after this one is merged.

[the remark this answers is not in the document]
Comment (SimonFrings) [reply composer.json]: I removed the whole autoload topic from this PR now and focused on adding the GitHub actions workflow, adding PHPUnit as a dev dependency and cleaning up a few minor things. This aligns the overall project structure with our other projects and lays the foundation to build additional PRs on top of this (autoloading, PHP versions, documentation, etc.)

**Review by clue — APPROVED** [review]
@SimonFrings Thanks for looking into this, changes LGTM! :shipit:

**Pull request merged**